### ISSUE TITLE
loop and lazy.checkInView are not compatible

### ISSUE THREAD

**Issue by ScottRoach** [nolimits4web/swiper#4374]
- Swiper Version: **6.5.0**
- Platform/Target and Browser Versions: **macOS 10.15.7** /  **Chrome 89.0.4389.90**

## What You Did

Enabling both `loop: true` and `lazy: { checkInView: true}` makes `checkInView` not work as expected.

I've forked the [Lazy Load Images Demo](https://swiperjs.com/demos/370-lazy-load-images.html) in Code Sandbox and have a working example here: 

https://codesandbox.io/s/swiper-lazy-load-images-forked-n0wn6?file=/index.html

Note that there will be a console error when the image loads prior to scrolling (or you can use the network tab to verify).

Not sure if this is a bug or a lack of documentation because of [this code](https://github.com/nolimits4web/swiper/blob/master/src/components/lazy/lazy.js#L247-L249). I figured it would be best to ask and either have the bug fixed or have documentation mention that these features are incompatible (or pointed to where I missed it).

## Expected Behavior

First Image loads when it is scrolled into the browsers viewport

## Actual Behavior

First image loads as the page loads

Cheers!

**Comment by Antimated** [comment]
I even have this issue with the loop set to true. Hopefully a fix comes soon.

**Comment by malininss** [comment]
I have the same problem. This is critically for me, is there some hacks for fix this?

**Comment by Pixelous** [comment]
@nolimits4web привет)

Можешь починить, пожалуйста?

**Comment by Pixelous** [comment]
What if using IntersectionObserver?

```
preloadImages: false,
lazy: false,
```

When observing using:

`swiper.params.lazy = { loadPrevNext: true };`

And so we can set rootMargin as well.

**Comment by Pixelous** [comment]
What I have found so far:

In https://github.com/nolimits4web/swiper/blob/master/src/modules/lazy/lazy.js#L243-L251 find

```
on('init', () => {
    if (swiper.params.lazy.enabled && !swiper.params.loop && swiper.params.initialSlide === 0) {
      if (swiper.params.lazy.checkInView) {
        checkInViewOnLoad();
      } else {
        load();
      }
    }
  });
```

on lines 243-251 and replace with:

```
on('init', () => {
    if (swiper.params.lazy.checkInView) {
      checkInViewOnLoad();
    } else {
      load();
    }
  });
```
No need to check:

`swiper.params.lazy.enabled && !swiper.params.loop && swiper.params.initialSlide === 0`

Need to check `checkInViewOnLoad` function further on lines 188-236:

https://github.com/nolimits4web/swiper/blob/master/src/modules/lazy/lazy.js#L188-L236

as looks inView var is always true even if the slider element is not in window view.

Any thoughts?

**Comment by Pixelous** [comment]
I have fixed it) Created a new issue here [#4900](https://github.com/nolimits4web/swiper/issues/4900)

**Comment by mskarimi** [comment]
> I have fixed it) Created a new issue here [#4900](https://github.com/nolimits4web/swiper/issues/4900)

is it work on v6?
I am using v6.8.4 and i have same problem

**Comment by Antimated** [comment]
> > I have fixed it) Created a new issue here [#4900](https://github.com/nolimits4web/swiper/issues/4900)
> 
> 
> 
> is it work on v6?
> 
> I am using v6.8.4 and i have same problem

No only works in v7
